src/server/scripts/Commands/cs_misc.cpp

```diff
@@ -1386,7 +1386,31 @@ class misc_commandscript : public CommandScript
 
         if (!target || !target->IsConnected())
         {
-            return false;
+            if (handler->HasLowerSecurity(nullptr, target->GetGUID()))
+                return false;
+
+            ObjectGuid::LowType guid = sCharacterCache->GetCharacterGuidByName(target->GetName()).GetCounter();
```

Comment:
You don't need to access the cache at all here, you already got the guid info

```cpp
                CharacterDatabasePreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_CHAR_HOMEBIND);
                stmt->SetData(0, player->GetGUID().GetCounter());
                PreparedQueryResult resultDB = CharacterDatabase.Query(stmt);
```

Reply:
OK. I'll check it out. Thank you.